libavdevice/dshow.c

```diff
@@ -311,6 +343,39 @@ dshow_cycle_devices(AVFormatContext *avctx, ICreateDevEnum *devenum,
     return 0;
 }
 
+static int dshow_get_device_list(AVFormatContext *avctx, AVDeviceInfoList *device_list)
+{
+    struct dshow_ctx *ctx = avctx->priv_data;
+    ICreateDevEnum *devenum = NULL;
+    int r;
+    int ret = AVERROR(EIO);
+
+    if (!device_list)
+        return AVERROR(EINVAL);
+
+    CoInitialize(0);
```

Comment:
expensive if we do this too often (maybe have one thread where we do com/dshow stuff?)

Reply:
yeh we can probably limit the amount of calls to this code when we create `FFmpegDevice` instances on our end - instead of calling `QueryForDevices()` we could refactor it out into a separate thing like we have on this branch https://github.com/SBGSports/SBGSource/blob/2b0efd1cf3c52149f279f017c23300418486e94a/RaceMedia/ffmpeg_device_list.cpp